maint_tools/check_gha_workflow.py

```diff
@@ -0,0 +1,263 @@
+    )
+
+    fig.update_xaxes(dtick="M1", tickformat="%b\n%Y")
+    fig.update_layout(autosize=True, width=1000, height=700)
+
+    fig.write_image(OUTPUT_FILE.with_suffix(".png"))
+    fig.write_html(OUTPUT_FILE.with_suffix(".html"))
+
+
+def set_os(x: str) -> str:
+    """Detect which OS the job was run on."""
+    if "ubuntu" in x:
+        return "ubuntu"
+    elif "windows" in x:
+        return "windows"
+    elif "macos" in x:
+        return "macos"
+    else:
+        return "n/a"
+
+
+def set_python_version(x: str) -> str:
+    """Detect which python version the job was run on."""
+    return next(
+        (
+            version
+            for version in ["3.8", "3.9", "3.10", "3.11", "3.12"]
+            if version in x
+        ),
+        "n/a",
+    )
+
+
+def set_dependencies(x: str) -> str:
+    """Detect which set of dependencies was used for the run."""
+    return next(
+        (
+            dependencies
+            for dependencies in ["pre-release", "no plotting", "no plotly"]
+            if dependencies in x
+        ),
+        "latest" if "latest dependencies" in x else "n/a",
+    )
+
+
+def get_auth(username: str, token_file: Path) -> None | tuple[str, str]:
+    """Get authentication with token."""
+    token = None
+
+    if token_file.exists():
+        with open(token_file) as f:
+            token = f.read().strip()
+    else:
+        warnings.warn(f"Token file not found.\n{str(token_file)}")
+
+    return None if username is None or token is None else (username, token)
+
+
+def get_runs(
+    workflow_id: str,
+    auth: None | tuple[str, str] = None,
+    page: int = 1,
+    include_failed_runs: bool = True,
+) -> list[dict[str, Any]]:
+    """Get list of runs for a workflow.
+
+    Restricted to:
+    - main branch
+    - completed runs
+    """
+    status = "completed"
+    branch = "main"
+
+    source = f"https://api.github.com/repos/{USER}/{REPO}/actions/workflows"
+    query = f"?per_page=100&status={status}&branch={branch}&page={page}"
+    url = f"{source}/{workflow_id}/runs{query}"
+
+    print(f"pinging: {url}")
+
+    content = handle_request(url, auth)
+
+    if not content.get("workflow_runs"):
+        return []
+    if include_failed_runs:
+        return [
+            i
+            for i in content["workflow_runs"]
+            if i["conclusion"] in ["success", "failure"]
+        ]
+    return [
+        i for i in content["workflow_runs"] if i["conclusion"] == "success"
+    ]
+
+
+def handle_request(url: str, auth: None | tuple[str, str]):
+    """Wrap request."""
+    if isinstance(auth, tuple):
+        response = requests.get(url, auth=auth)
+    elif isinstance(auth, dict):
+        response = requests.get(url, headers=auth)
+
+    if response.status_code != 200:
+        raise RuntimeError(
+            f"\n Got code: {response.status_code}.\n"
+            f" With response: {response.json()}"
+        )
+
+    return response.json()
+
+
+def udpate_jobs_data(
```

Comment:
Will also need to update the associated call(s)

```suggestion
def update_jobs_data(
```

Reply:
good catch
will fix the calls